### DISCUSSION TITLE
Add support for changing `variantColorResolver` based on `colorScheme`

### DISCUSSION THREAD

**Discussion by kirilllyalin** [mantinedev/mantine#6093] [Feature requests]
Hello!

I was faced with the need to change the colors of certain component variants based on the current project color scheme, within the `variantColorResolver` function.

In the current project, some components variants have 2 presenations: for a dark color scheme and for a light one.

1. Perhaps this can be done with the current implementation of the library?
2. If not, then I suggest adding an override at the theme creation level.

For example: 

```
export const baseTheme = createTheme({
  ...themeProps,
  primaryShade: { dark: 9, light: 5 },
  variantColorResolver: { dark: darkThemeVariantColorResolver, light: lightThemeVariantColorResolver }
})
```

**Comment by rtivital** [comment]
It is not feasible. Color scheme is not determined by js value, it is defined by a data attribute on the root element. Switching resolver based on this value will break ssr.
